src/main/java/redis/clients/jedis/XReadArgs.java

```diff
@@ -2,40 +2,96 @@
 
 import redis.clients.jedis.exceptions.JedisDataException;
 
-import java.util.LinkedList;
-import java.util.List;
+import java.util.*;
 
 public class XReadArgs {
 
-    public final List<String> streams;
-    public final List<String> offsets;
-    public final int count;
-    public final boolean block;
-    public final long blockDuration;
+    public final List<String> streamsAndOffsets;
+    public int count;
+    public boolean block;
+    public long blockDuration;
+
+    /**
+     * Creates simple XReadArgs object.
+     * @param streamsAndOffsets List of stream names and offsets... stream names first, offsets second.
+     */
+    public XReadArgs(List<String> streamsAndOffsets) {
+        this.streamsAndOffsets = streamsAndOffsets;
+        this.count = 0;
+        this.block = false;
+        this.blockDuration = 0;
+    }
 
-    private XReadArgs(List<String> streams, List<String> offsets, int count, boolean block, long blockDuration) {
-        this.streams = streams;
-        this.offsets = offsets;
+    private XReadArgs(List<String> streamsAndOffsets, int count, boolean block, long blockDuration) {
+        this.streamsAndOffsets = streamsAndOffsets;
         this.count = count;
         this.block = block;
         this.blockDuration = blockDuration;
     }
 
+    public void setStreamsAndOffsets(List<String> streamsAndOffsets) {
+        this.streamsAndOffsets.clear();
+        if (streamsAndOffsets.size() < 50) {
+            // If the incoming value is small enough, let's opt to iterate with .add to copy it into the XReadArgs and
+            // save on the allocations.
+            for (String s : streamsAndOffsets) {
+                //noinspection UseBulkOperation
+                this.streamsAndOffsets.add(s);
+            }
+        } else {
+            this.streamsAndOffsets.addAll(streamsAndOffsets);
+        }
+    }
+
+    public String getKey() {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        int size = streamsAndOffsets.size();
+        int i = 0;
+        for (String s : streamsAndOffsets) {
+            if (i == size / 2) {
+                break;
+            }
+            stringBuilder.append(s);
+            i++;
+        }
+
+        return stringBuilder.toString();
+    }
+
+    public void setCount(int count) {
+        this.count = count;
+    }
+
+    public void setBlock(boolean block) {
+        this.block = block;
+    }
+
+    public void setBlockDuration(long blockDuration) {
+        this.blockDuration = blockDuration;
+    }
+
     public static Builder builder() {
         return new Builder();
     }
 
     public static class Builder {
 
-        private final List<String> streams = new LinkedList<>();
-        private final List<String> offsets = new LinkedList<>();
+        private final List<String> streams = new ArrayList<>();
```

Comment:
Not sure why this is not a linked list...

Reply:
Linked lists create an object for every item you add to them. Array lists don't